doc/ref_model/chapters/appendix-a.md

```diff
@@ -1,80 +1,80 @@
-2. Provide usable definitions of maturity levels for VNF software architecture between Physical-to-Virtual migration and “Cloud Native”.
-
-The goal is not to be prescriptive on how to re-architect existing or architect new applications but rather staying within scope of focusing on interface and interaction between applications and platform.
+## A.1 Guidelines for VNF Vendors
 
 <a name="A.2"></a>
-## A.2 Intro and Terminology
-(Summary status and trends of ETSI NFV and Cloud. Decoupling application from platform. Terminology used/introduced.)
+## A.2 Guidelines for CNF Vendors
 
 <a name="A.3"></a>
-## A.3 VNF Maturity Levels
-(And how they map to CNTT RAs)
-
-<a name="A.4"></a>
-## A.4 Links
-
-<a name="A.5"></a>
-## A.5 Hardware-Dependent Coding Policies
-
-As described in the Principles sections of RM Chapter 1, features that require hardware-dependent code in the workload are prohibited in CNTT compliant implementations. This principle is henceforth referred to as the "Abstraction Principle". Note, this is not the case for the Compute node host software (e.g., host OS). Within the Infra, hosts are expected to have software that is customized for the specific hardware equipped. However, the intent is that these software drivers and higher layers will abstract the Capabilities they enable, thereby exposing them with an open API. An example of exposing capabilities in this manner is implemented in the Virtio family of APIs. This requirement is in support of VNF abstraction and portability of VNFs across the Infra landscape.
+## A.3 VNF Transitional Generations
 
-CNTT realizes there are implications to this and limitations to the ability to live by the Abstraction Principle. A textbook example of a Capability that transgresses this principle, is SR-IOV. Other, less notable, yet very important examples include vGPUs and other acceleration hardware, such as FPGA. A less obvious, yet critically important example is the VNF program(s) itself. As workloads execute native microarchitecture opcodes, those instruction sets effectively constitute an ABI (Application Binary Interface). Additionally, these programs may or may not, attempt to execute vendor-specific extensions to standard instruction sets, such as x86 or ARM.
+**Non-Conformant**: (Non-Conformant): None are aligned.
 
-Solving the problems associated with implementing the Abstraction Principle is a work in progress. CNTT has not solved all of the associated problems, nor has the industry. As technology evolves and more designs incorporate cloud native concepts, these problems will be addressed. This appendix is specifically intended to provide CNTT policies to manage these situations as they exist today,   and their exceptions and transitions, as the technology around and supporting the Abstraction Principle matures.
+**Gen1**: (Consistent): Going to the right direction but still lots of work to do.
```

Comment:
Should we classify Generations as Consistent | Conformant etc. or should we define Generations with software versions and allowed exceptions. We can also define degree of compliance with labels such as Conformant | Consistent etc. 

For example, RA-1 Generation 1 (as it stands) can be based on OpenStack Pike version and the API versions specified in the current RA-1 Chpater05; other software versions for libvirt etc. can also be specified for completeness. Additionally, RA-1 Generation 1 can specify an acceptable exception for the use of SR-IOV and qualify for "Compliant" while without SR-IOV to qualify for "Fully Compliant." 

RA-1 Generation 2 could be based on OpenStack Stein release, for example, with suitable changes to Chapter 5 and other software versions.

Reply:
This is great point. I do suggest we get some fundamental agreement on these. I will add to next TSC call.